powerauth-java-server/src/test/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/OperationTemplateServiceBehaviorTest.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * PowerAuth Server and related software components
+ * Copyright (C) 2023 Wultra s.r.o.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.getlime.security.powerauth.app.server.service.behavior.tasks;
+
+import com.wultra.security.powerauth.client.model.enumeration.SignatureType;
+import com.wultra.security.powerauth.client.model.request.OperationTemplateCreateRequest;
+import io.getlime.security.powerauth.app.server.service.exceptions.GenericServiceException;
+import io.getlime.security.powerauth.app.server.service.model.ServiceError;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Test for {@link OperationTemplateServiceBehavior}.
+ *
+ * @author Jan Pesek, jan.pesek@wultra.com
+ */
+@SpringBootTest
+class OperationTemplateServiceBehaviorTest {
+
+    @Autowired
+    private OperationTemplateServiceBehavior service;
+
+    @Test
+    void testDuplicateOperationTemplateCreation() throws Exception {
+        final String templateName = "login";
+
+        service.createOperationTemplate(createOperationTemplateCreateRequest("login"));
```

Comment:
Didn't you want to use `templateName`? Anyway, what is the reason to test both repository and service?
```suggestion
        service.createOperationTemplate(createOperationTemplateCreateRequest(templateName));
```

Reply:
> Anyway, what is the reason to test both repository and service?

The `OperationTemplateServiceBehavior.createOperationTemplate()` checks uniqueness on service level by querying a template by its name first. If a template with the same name already exists, it throws an exception without triggering database constraint. Although it means the template creation using the API is not influenced by this PR, I wanted to cover it with a test, as it is closely related.

This issue address direct database insertion, which must be handled by the database constraint and is therefore tested on the repository level.